apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx

```diff
@@ -0,0 +1,356 @@
+        regionalPartnerWorkshopCount: data.workshops ? data.workshops.length : 0
+      });
+
+      // Update state with all the partner workshop data to display
+      this.setState({
+        loadingPartner: false,
+        partnerWorkshops: data.workshops
+      });
+    });
+  }
+
+  loadAlternateWorkshops() {
+    const url = `/api/v1/pd/regional_partner_workshops?${$.param(this.getWorkshopParams())}`;
+    this.loadAlternateWorkshopsRequest = $.ajax({
+      method: 'GET',
+      url: url,
+      dataType: 'json'
+    }).done(data => {
+      this.loadAlternateWorkshopsRequest = null;
+      const alternateWorkshops = data.reduce((workshops, partner) => (
+        partner.id !== this.props.data.regionalPartnerId
+          ? workshops.concat(partner.workshops)
+          : workshops
+      ), []);
+
+      this.setState({
+        loadingAlternateWorkshops: false,
+        alternateWorkshops
+      });
+    });
+  }
+
+  getWorkshopParams() {
+    const course = {
+      [PROGRAM_CSD] : 'CS Discoveries',
+      [PROGRAM_CSP] : 'CS Principles'
+    }[this.props.data.program];
+
+    return {
+      course,
+      subject: '5-day Summer'
+    };
+  }
+
+  isUnableToAttendAssignedWorkshop = (data = this.props.data) => data.regionalPartnerId && (
+      data.ableToAttendSingle === UNABLE_TO_ATTEND ||
+      (
+        data.ableToAttendMultiple &&
+        data.ableToAttendMultiple.length === 1 &&
+        data.ableToAttendMultiple[0] === NO_EXPLAIN
+      )
+    );
+
+  /**
+   * @override
+   */
+  handleChange(newState) {
+    const data = {...this.props.data, ...newState};
+    if (this.isUnableToAttendAssignedWorkshop(data) && !this.state.alternateWorkshops && !this.loadAlternateWorkshopsRequest) {
+      this.loadAlternateWorkshops();
+    }
+
+    super.handleChange(newState);
+  }
+
+  handleUnderstandFeeChange = event => {
+    this.handleChange({
+      understandFee: event.target.checked
+    });
+  };
+
+  handleConsiderForFundingChange = event => {
+    this.handleChange({
+      considerForFunding: event.target.checked
+    });
+  };
+
+  renderAbleToAttendSingle() {
+    const options = [
+      "Yes, I'm able to attend",
+      UNABLE_TO_ATTEND
+    ];
+    const textFieldMap = {[UNABLE_TO_ATTEND]: 'explain'};
+    return this.dynamicRadioButtonsWithAdditionalTextFieldsFor(
+      "ableToAttendSingle",
+      options,
+      textFieldMap
+    );
+  }
+
+  renderAssignedWorkshopList() {
+    if (this.state.loadingPartner) {
+      return null;
+    }
+
+    if (!this.props.data.regionalPartnerId) {
+      return (
+        <div>
+          There currently is no Regional Partner in your area.
+          If a seat opens in the program, we will invite you to a TeacherCon and provide you with more details.
+        </div>
+      );
+    } else if (this.props.data.regionalPartnerGroup === 3 && this.props.data.regionalPartnerWorkshopCount === 0) {
+      // TODO: find TC based on G3 partner match
```

Comment:
I would stick your name in this TODO as well if you're going to be doing that work